### ISSUE TITLE
Bug: CodeIgniter\Controller::validateData will throw error if first paramter is not an empty array for file upload validation

### ISSUE THREAD

**Issue by hfdzz** [codeigniter4/CodeIgniter4#8874]
### PHP Version

8.1

### CodeIgniter4 Version

4.5.1

### CodeIgniter4 Installation Method

Composer (using `codeigniter4/appstarter`)

### Which operating systems have you tested for this bug?

Windows

### Which server did you use?

apache

### What happened?

**Controller**:
```php
$files = $this->request->getFiles();

$rules = [
    'test' => 'uploaded[test]|max_size[test,1024]|ext_in[test,png,jpg,jpeg,pdf]'
];

if (!$this->validateData($files, $rules)) {
    return redirect()->back()->withInput()->with('errors', $this->validator->getErrors());
}

return 'File uploaded successfully';
```

**Error Response**:
`TypeError: CodeIgniter\Validation\FileRules::uploaded(): Argument #1 ($blank) must be of type ?string, CodeIgniter\HTTP\Files\UploadedFile given, called in {SYSTEMPATH}\Validation\Validation.php on line 336 `

### Steps to Reproduce

Form with file input and validation in controller using `Controller::validateData` (exactcly the same as I did).

### Expected Output

Validation run accordingly without error.

### Anything else?

`Controller::validate` works just fine.
The documentation gives example using `validateData` without **explicitly** saying to use empty array for `$data` parameter (this works just fine as well).

![image](https://github.com/codeigniter4/CodeIgniter4/assets/80891279/5aa4a993-1c24-495f-b5f1-e4437953090b)
The problem just show up after i update my CodeIgniter 4.4.4 to 4.5.1.
After taking a look at system files, i noticed a new line `declare(strict_types=1);` that may cause the error due to type error (which was not there on 4.4.4). 

Not sure if this is intentional, but took me a while to figure this out.

**Comment by kenjis** [comment]
You should not pass the `$files` to `validateData()`. 
Your code is a misuse, and the error message is correct.

It may be difficult to understand, but upload files are not included in the data passed to `validateData()`.
This is because the file validation rules validate the uploaded file in `$this->request` directly. You have passed unnecessary data to the file validation rules. So, since v4.5, the type checking is more strict (`declare(strict_types=1)`) and an error occurs.

In conclusion, you should pass `[]` to `validateData()` if you validate only uploaded files.
If you have `$_POST` data, you should pass it to `validateData()`.

**Comment by kenjis** [comment]
We use GitHub issues to track BUGS and to track approved DEVELOPMENT work packages. We use our [forum](http://forum.codeigniter.com/) to provide SUPPORT and to discuss FEATURE REQUESTS.

**Comment by kenjis** [comment]
I send a PR to improve the docs: #8876

**Comment by hfdzz** [comment]
I thought this is unintentional since it works before 4.5 and i didn't look at the system file thorougly.
Thanks
